From a006fc83b451b9b988f19253d23845886de0db0a Mon Sep 17 00:00:00 2001
From: oliverwen <oliverwen@google.com>
Date: Thu, 14 Jul 2016 09:41:20 -0700
Subject: [PATCH] CHROMIUM: smart antenna algorithm: bugfix: initialize rx_ant
 to be default ant value

 - fix the bug of initial rx_ant value
   bug: initial rx_ant = 0, not the default ant.
   fix: initialize rx_ant with default ant value

TEST=when AP is rebooted, enable smart antenna log by
echo 0x4000 > /sys/module/ath10k_core/parameters/debug_mask
read initial rx_ant configuration in /var/log/messages as
"Configure Rx antenna to X", where "X" should be the default ant value
if "Configure tx antenna for <mac_address> to Y" Y=default ant value.

BUG=chrome-os-partner:51964
CQ-DEPEND=CL:341563

Signed-off-by: oliverwen <oliverwen@google.com>
(cherry picked from commit 9f0df56c12bf83c83cb46fc6dcc47ee639073341)
(source: wireless-3.18)

Change-Id: I9a8bde72cc6985730a517e272c7bd71e8724447c
Reviewed-on: https://chromium-review.googlesource.com/341564
Commit-Ready: Oliver Wen <oliverwen@chromium.org>
Tested-by: Oliver Wen <oliverwen@chromium.org>
Reviewed-by: Yixiang Li <yixiang@google.com>
---
 drivers/net/wireless/ath/ath10k/smart_ant.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath10k/smart_ant.c b/drivers/net/wireless/ath/ath10k/smart_ant.c
index 886e79084a3..842648471f3 100644
--- a/drivers/net/wireless/ath/ath10k/smart_ant.c
+++ b/drivers/net/wireless/ath/ath10k/smart_ant.c
@@ -1818,6 +1818,7 @@ int ath10k_smart_ant_sta_connect(struct ath10k *ar,
 			ATH10K_SMART_ANT_TRAIN_TRIGGER_RX;
 	smart_ant_sta->train_info.train_state =
 			ATH10K_SMART_ANT_STATE_PRETRAIN;
+	info->rx_ant = info->default_ant;
 
 	if (sta->deflink.vht_cap.vht_supported)
 		smart_ant_sta->wmode = ATH10K_WIRELESS_MODE_VHT;
-- 
2.43.0

